firmware/src/ledModifier.cpp

```diff
@@ -0,0 +1,93 @@
+#include "ledModifier.hpp"
+
+#include <cmath>
+
+#include "pico/stdlib.h"
+
+static HSV *led_array;
+static long *tick;
+const int nr_of_leds = 5;  // OBS, should be inherrited once linked
+
+// used to silence intellisenses weird hate for M_PI
+static const double pi = 3.14159265359;
+
+void setPointers(HSV *led_array_pointer, long *tick_pointer) {
+	led_array = led_array_pointer;
+	tick = tick_pointer;
+}
+
+void setColour(HSV colour, uint8_t bitmask) {
+	for (int i = 0; i < nr_of_leds; i++) {
+		led_array[i] = colour;
+	}
+}
+
+void changeHue(HSV &led, int amount) {
+	led.hue = ((led.hue + amount) + 255) % 255;	 // looping
```

Comment:
I notice `255` being used in a few places. This is what we call a "magic number" which can (or should) be stored as a const value with a useful name to make its purpose clearer to the reader.

https://www.inf.unibz.it/~calvanese/teaching/04-05-ip/lecture-notes/uni04/node15.html

Reply:
Good point. Implemented.